src/mlang/m_frontend/mast_to_mir.ml

```diff
@@ -255,10 +256,7 @@ let var_or_int_value (ctx : translating_context) (l : Mast.literal Pos.marked) :
       try
         (* We look up the value of the variable, which has to be const *)
         let name = Mast.get_variable_name v in
-        Mir.VariableMap.find
-          (get_var_from_name ctx.idmap (Pos.same_pos_as name l) ctx.exec_number
-             false)
-          ctx.int_const_values
+        ConstMap.find name ctx.const_map |> Pos.unmark |> int_of_float
```

Comment:
We could check that there is no loss of precision when calling `int_of_float`

Reply:
You mean in case of a non-integer value ? Doesn't the M semantics truncate anyway for array accesses ?